src/main/java/dev/blachut/svelte/lang/parsing/html/SvelteHtmlLexer.kt

```diff
@@ -56,4 +64,15 @@ class SvelteHtmlLexer : HtmlLexer(InnerSvelteHtmlLexer(), false) {
     override fun isHtmlTagState(state: Int): Boolean {
         return state == _SvelteHtmlLexer.START_TAG_NAME || state == _SvelteHtmlLexer.END_TAG_NAME
     }
+
+    companion object {
+        fun getSvelteScriptContentProvider(mimeType: String?): HtmlScriptContentProvider? {
+            if (mimeType == "ts" || mimeType == "typescript") {
+                //todo here we should use language extension similar to SvelteJSLanguage, instead of the original lang
+                return LanguageHtmlScriptContentProvider.getScriptContentProvider(JavaScriptSupportLoader.TYPESCRIPT)
```

Comment:
@tomblachut the main issue here is that we cannot directly add ``SvelteTypeScriptParser`` similar to ``SvelteJSParser`` because constructors of ``TypeScriptStatementParser`` and other classes are package-protected :(

Reply:
I can try some solutions:
1. Add a delegate parser that extends ES6Parser
2. Add an inheritor inside ``com.intellij.lang.javascript.ecmascript6.parsing`` (but I am not sure that it will work because of IDE custom class loading)

Reply:
@anstarovoyt I find 2nd solution really confusing for new contributors - IDK if it's something accepted in Java/Kotlin land, so my opinion is strictly first impression.

Delegate parser could be later maybe incorporated into WS (or you can change access modifier) so even if it's a bit harder to maintain for now it could be later refactored and removed from this repo.

We could also discuss whether custom langs are really necessary as long as stores and reactive statements are still WIP, currently I have no idea how to implement them without parser overrides

Reply:
>We could also discuss whether custom langs are really necessary as long as stores and reactive statements are still WIP, currently I have no idea how to implement them without parser overrides

At this moment I see only one reason why we need a separate psi for such elements: "unresolved element" warning. But I think the problem can be solved using implicit elements, I will take a look. Do you see any other problems if we don't process reactive statements as real variable declarations?

Reply:
if you don't see any issues with this PR, I think it is ready-to-merge as a basic ts support in svelte.

Reply:
I finally had a chance to take a look at reactive declarations. First of all, I'm no longer sure #97 is even correct. Svelte has some edge cases around reactive declarations, they are not strict variable statements as far as I understand.

Now I'm looking into another approach, to cover them similarly to `SvelteJSReferenceExpression` from stores implementation, i.e. replacing `PsiElement` in `SvelteJSPsiBuilder`. `TypeScriptParser` ctor is public, so this would work without delegates/hacks etc.

`SvelteJSLabeledStatement.processDeclarations` needs to be overriden, but it's not all. Resolving from template to script needs some more code.
- `processDeclarations` in script tag calls
- `JSEmbeddedContentImpl.processDeclarations` calls
`JSResolveUtil.processDeclarationsInScope(this, processor, state, lastParent, place)`
- `JSScopeNamesCache.getNamedElementsInScope(scope, requiredName, false)`
- `ScopeNamesCache.JSNamesCacheVisitor` — it handles `visitJSVarStatement` but not `visitJSLabeledStatement`

My idea is to implement both `JSLabeledStatement` and `JSVarStatement` by `SvelteJSLabeledStatement` OR rework resolving to skip `JSEmbeddedContentImpl.processDeclarations` and directly process each statement in `JSEmbeddedContent`, not sure what is better right now.

Anyway, stores will work with patched `TypeScriptParser`, reactive declarations are subject to more investigation, so basic TS is within hands reach. I need to merge some PRs and publish a release this week, should've done it week ago already.

Reply:
@tomblachut any updates?

Reply:
@anstarovoyt You can take a peek at #163, I've recently took different approach to (initial) support for reactive declarations. It favours no inspections instead of confusing/wrong inspections and it does not require parser changes.

---

TS support will require couple of small additions on top of parser
* $$props etc resolving
* $storeSubscriptions - right now there is one way support, TS doesn't support it

SvelteJSReferenceExpressionResolver handles both of the above.

I think this will require 
```
public class SvelteTSLanguage extends JSLanguageDialect implements DependentLanguage {
    public static final SvelteTSLanguage INSTANCE = new SvelteTSLanguage();

    private SvelteTSLanguage() {
        super("SvelteTS", DialectOptionHolder.TS, JavaScriptSupportLoader.TYPESCRIPT);
    }
}
```
and mirrored `SvelteTSSpecificHandlersFactory` & `SvelteTSReferenceExpressionResolver`.  Same for `SvelteJSScriptContentProvider` & dedicated `IElementType`.

I hope that this SvelteTSLanguage won't break other things.

Having said that I' not sure what is best to do right now:
1. Merge this anyway and provide at least something for the users. They will be angry (at me) that above don't work.
2. First release next version with changes already in master, #146 & #163 and then finish missing pieces for TS.

Anyway I'm really happy that this reactive declarations vs TS problem is coming to an end.

Right now I'm going to sleep 😅

Reply:
>Merge this anyway and provide at least something for the users. They will be angry (at me) that above don't work.
or
>First release next version with changes already in master, #163 and then finish missing pieces for TS.

I think I can fix the rest issues with the ts support in 1-2 days after merging #146 & #163 so we can do it all together

Reply:
by the way. 
>$$props etc resolving
>$storeSubscriptions - right now there is one way support, TS doesn't support it
This PR doesn't inject TypeScript into html-specific things (only to the script tag), so everything should still work

Reply:
Actually you can use $subscriptions in script tag also. 

I'm not sure how it works in terms of unwrapping the type, but for JS example please take a look at
https://svelte.dev/repl/130bc99a24d442afb27f0fd8f9ddec86?version=3.29.0

Reply:
That's the biggest difference from Vue IMO, you need custom languages for everything, not only expressions

Reply:
oh, I see. Thanks. I will try to fix it asap

Reply:
Okay, thanks! I've merged #146 already and will merge #163 shortly.

Reply:
@tomblachut I've updated the PR

Reply:
@anstarovoyt thanks I'll take a look in the evening